### PR TITLE
Generate and display data in Swipeable

### PR DESCRIPTION
**Pull request by avine**
- Helper function to convert data from CSV to Object
- Generate random dataset 
- Testing npm package "react-deck-swiper"

### CODE REVIEW

File: public/page-data/cards/page-data.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"componentChunkName":"component---src-pages-cards-js","path":"/cards/","result":{"pageContext":{}},"staticQueryHashes":[]}
```
Comment (blacroix) [inline public/page-data/cards/page-data.json]: I removed public dir in another PR, public dir isn't required.

File: src/data/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import data from "./data";
```
Comment (blacroix) [inline src/data/index.js]: May we use TypeScript?

File: src/pages/index.js [hunk 3]
```diff
@@ -1,150 +1,76 @@
-import * as React from "react"
+import * as React from "react";
+import { Swipeable, direction } from "react-deck-swiper";
+import getData from "../data";
 
-// styles
-const pageStyles = {
-  color: "#232129",
-  padding: "96px",
+const container = {
```
Comment (blacroix) [inline src/pages/index.js]: May we use SCSS modules?

**Review by blacroix — CHANGES_REQUESTED** [review]

File: src/data/data.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
+export default [
```
Comment (blacroix) [inline src/data/data.js]: Gatsby uses a datasource system, you don't have to map csv to json yourself: https://whatjackhasmade.co.uk/using-a-csv-file-as-a-data-source-in-gatsby

**Review by blacroix — CHANGES_REQUESTED** [review]

File: src/data/index.js (hunk 2 above)
Comment (avine) [reply src/data/index.js]: Of course, but i don't know how to setup this ?

File: src/pages/index.js (hunk 3 above)
Comment (avine) [reply src/pages/index.js]: Of course, this is just a dummy rendering. This PR is about the swipe effect and the data management.

File: src/data/data.js (hunk 4 above)
Comment (avine) [reply src/data/data.js]: Thx. :)

File: src/data/index.js (hunk 2 above)
Comment (blacroix) [reply src/data/index.js]: You don't have to, just rename your file to `*.tsx` and use TS ;)

File: src/data/data.js (hunk 4 above)
Comment (avine) [reply src/data/data.js]: If i need to transform the CSV, how do i do that ? :(

Under the hood, the package `gatsby-transformer-csv` uses `node-csvtojson`.

https://github.com/Keyang/node-csvtojson#hook--transform

Comment (blacroix) [reply src/data/data.js]: I think the point here is to easily mount the csv and use graphql then to read it; you can after map the result to whatever you want 🤔

File: public/page-data/cards/page-data.json (hunk 1 above)
Comment (avine) [reply public/page-data/cards/page-data.json]: ok

File: src/data/data.js (hunk 4 above)
Comment (blacroix) [reply src/data/data.js]: Doc is here: https://www.gatsbyjs.com/docs/content-and-data/

Comment (avine) [reply src/data/data.js]: Yes. But i might be better to have the data formatted from the start.

Comment (avine) [reply src/data/data.js]: With your solution, the non-formatted data will be packaged in the distribution and will be fixed at runtime by a `.map()` function.

Comment (avine) [reply src/data/data.js]: In other words, how to transform the CSV at built time ?

Comment (blacroix) [reply src/data/data.js]: I don't know :) I suggest to use Gatsby way to consume data source but if it's not appropriate we may do it your way.

File: src/data/index.js (hunk 2 above)
Comment (avine) [reply src/data/index.js]: ok

File: src/data/data.js (hunk 4 above)
Comment (avine) [reply src/data/data.js]: Ok, for now, let's do it the Gatsby way :)

**Review by blacroix — APPROVED** [review]

**Pull request merged**